syntax/go.vim

```diff
@@ -89,12 +89,30 @@ if !exists("g:go_highlight_generate_tags")
   let g:go_highlight_generate_tags = 0
 endif
 
+let s:fold_block = 1
+let s:fold_import = 1
+let s:fold_varconst = 1
+if exists("g:go_fold_enable") && type(g:go_fold_enable) == type([])
+  if index(g:go_fold_enable, 'block') == -1
+    let s:fold_block = 0
+  endif
+  if index(g:go_fold_enable, 'import') == -1
+    let s:fold_import = 0
+  endif
+  if index(g:go_fold_enable, 'varconst') == -1
+    let s:fold_varconst = 0
+  endif
+endif
+
 syn case match
 
-syn keyword     goDirective         package import
-syn keyword     goDeclaration       var const
+syn keyword     goPackage           package
+syn keyword     goImport            import    contained
+syn keyword     goVarConst          var const contained
```

Comment:
This changes the semantics of the syntax settings. In go const, var and type all are declarations, so `goDeclaration` is the correct  setting here.  Actually things like `import` or `func` are also declarations. I'm not sure why they were added as `goDirective` back then.  If you need to access them, I think the better way would be using `goVar` and `goConst`, instead of combining them. At least that would give it a correct meaning (even if it's more fine grained now)